ui/app/helpers/parse-pki-cert.js

```diff
@@ -1,24 +1,53 @@
 import { helper } from '@ember/component/helper';
-import { pki } from 'node-forge';
+import * as asn1js from 'asn1js';
+import { fromBase64, stringToArrayBuffer } from 'pvutils';
+import { Certificate } from 'pkijs';
 
 export function parsePkiCert([model]) {
   // model has to be the responseJSON from PKI serializer
   if (!model.certificate) {
     return;
   }
   let cert;
-  // node-forge cannot parse EC (elliptical curve) certs
-  // set canParse to false if unable to convert a Forge cert from PEM
   try {
-    cert = pki.certificateFromPem(model.certificate);
+    let cert_base64 = model.certificate.replace(/(-----(BEGIN|END) CERTIFICATE-----|\n)/g, '');
+    let cert_der = fromBase64(cert_base64);
+    let cert_asn1 = asn1js.fromBER(stringToArrayBuffer(cert_der));
+    cert = new Certificate({ schema: cert_asn1.result });
   } catch (error) {
+    console.log('Error parsing certificate:', error, model.certificate);
     return {
       can_parse: false,
     };
   }
-  const commonName = cert?.subject.getField('CN') ? cert.subject.getField('CN').value : null;
-  const expiryDate = cert?.validity.notAfter;
-  const issueDate = cert?.validity.notBefore;
+
+  // We wish to get the CN element out of this certificate's subject. A
+  // subject is a list of RDNs, where each RDN is a (type, value) tuple
+  // and where a type is an OID. The OID for CN can be found here:
+  //
+  //    http://oid-info.com/get/2.5.4.3
+  //    https://datatracker.ietf.org/doc/html/rfc5280#page-112
+  //
+  // Each value is then encoded as another ASN.1 object; in the case of a
+  // CommonName field, this is usually a PrintableString, BMPString, or a
+  // UTF8String. Regardless of encoding, it should be present in the
+  // valueBlock's value field if it is renderable.
+  const commonNameOID = '2.5.4.3';
+  const commonNames = cert?.subject?.typesAndValues
+    .filter((rdn) => rdn?.type === commonNameOID)
+    .map((rdn) => rdn?.value?.valueBlock?.value);
```

Comment:
In case anyone was wondering, this does appear to be the correct way to parse the RDN's value: https://github.com/PeculiarVentures/PKI.js/blob/master/src/AttributeTypeAndValue.js#L213-L214